sklearn/linear_model/least_angle.py

```diff
@@ -1437,7 +1438,7 @@ def fit(self, X, y, copy_X=True):
 
         self.alphas_ = alphas_
         with np.errstate(divide='ignore'):
-            self.criterion_ = n_samples * np.log(mean_squared_error) + K * df
+            self.criterion_ = n_samples * mean_squared_error/sigma + K * df
```

Comment:
if you want to divide everything by n then we should have

mean_squared_error / sigma2 + K * df / n_samples

It was a long while ago but I remember getting the same results as the R package from the authors when I used the corrected version.